operation/operation.c

```diff
@@ -5,9 +5,9 @@
 
 #include "operation.h"
 
-static volatile bool opto_fall = false;
-static volatile bool opto_rise = false;
-static volatile bool piezo_falling_edge = false;
+static volatile bool interrupt_flags[] = {
+        false, false, false
+};
```

Comment:
this is cleaner than before. I think struct would give clearer structure than array. then you don't need enum either. I would prefer using struct.

Reply:
Hmm... I'm unsure how they would be called/handled without enum ;
removing the enum would require us to call the flags with 0, 1 and 2 -- which works, but isn't very clear ; hence the enum declaration.

I'm interested if you can do it without enums, but am unsure how struct could replace them.

Reply:
yeah, you're right. struct in this case maybe wont make it clear enough. Let's go with this!